lib/masamune/abstract_syntax_tree/prism/node_extensions.rb

```diff
@@ -0,0 +1,74 @@
+module Prism
+  class Node
+    def comment? = false
+
+    def self.node_predicate(name, body = -> { true })
+      define_method(name, &body)
+      Node.define_method(name) { false } unless Node.respond_to?(name, false)
```

Comment:
Interesting, so the method is defined for the class that's calling it (i.e. - `node_predicate :method?` in `DefNode`), and then for the base Node class which all others inherit return `false` since the method doesn't exist for the other nodes yet. This works for me as well.

Reply:
It's somewhat annoying I couldn't just use `node_predicate` throughout, but when we'd hit the second or third of the same kind like `variable?`, we'd get method redefinition warnings on stdout.